WordPress/Classes/Services/BloggingPromptsService.swift

```diff
@@ -179,17 +179,45 @@ class BloggingPromptsService {
 
     // MARK: - Init
 
+    /// Initializes a service for blogging prompts.
+    ///
+    /// - Parameters:
+    ///   - contextManager: The CoreDataStack instance.
+    ///   - remote: When supplied, the service will use the specified remote service.
+    ///     Otherwise, a remote service with the default account's credentials will be used.
+    ///   - blog: When supplied, the service will perform blogging prompts requests for this specified blog.
+    ///     Otherwise, this falls back to the default account's primary blog.
     required init?(contextManager: CoreDataStackSwift = ContextManager.shared,
                    remote: BloggingPromptsServiceRemote? = nil,
                    blog: Blog? = nil) {
-        guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: contextManager.mainContext),
-              let siteID = blog?.dotComID ?? account.primaryBlogID else {
+        var remoteInstance = remote
+        var siteID: NSNumber? = nil
+
+        // if a blog exists, then try to use the blog's ID.
+        if let blog,
+           let blogContext = blog.managedObjectContext {
+            blogContext.performAndWait {
+                siteID = blog.dotComID
+            }
+        }
+
+        // fetch the default account and fall back to default values as needed.
+        contextManager.performQuery { mainContext in
+            guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: mainContext) else {
+                return
+            }
+            siteID = siteID ?? account.primaryBlogID
+            remoteInstance = remoteInstance ?? .init(wordPressComRestApi: account.wordPressComRestV2Api)
+        }
+
+        guard let siteID,
+              let remoteInstance else {
             return nil
         }
 
         self.contextManager = contextManager
         self.siteID = siteID
-        self.remote = remote ?? .init(wordPressComRestApi: account.wordPressComRestV2Api)
+        self.remote = remoteInstance
     }
```

Comment:
@dvdchr I'm wondering if it's possible to safely access both the `siteID` and the `WPAccount` instance using just the `mainContext`. cc @crazytonyli 

```swift
var remoteInstance = remote
var siteID: NSNumber? = nil

let context = contextManager.mainContext
context.performAndWait { 
     if let blogID = blog?.objectID  {
            let blogInContext = try? context.existingObject(with: blogID)
            siteID =   blogInContext?.dotComID
     }
     let account = try? WPAccount.lookupDefaultWordPressComAccount(in: context)  
     // ...
}

// Then we set `self` properties.
```

Reply:
Agreed. We could use `coreDataStack.performQuery` function (which uses the main context) as I suggested in https://github.com/wordpress-mobile/WordPress-iOS/pull/20854#discussion_r1228903792.

Reply:
Thanks for the input, folks 🙇🏼‍♂️. This should be addressed in https://github.com/wordpress-mobile/WordPress-iOS/commit/efdad6e4a7db74ac3503246eae8cab9f8f14f332.